src/v0/destinations/twitter_ads/transform.js

```diff
@@ -0,0 +1,227 @@
+const sha256 = require('sha256');
+const crypto = require('crypto');
+const oauth1a = require('oauth-1.0a');
+const {
+  constructPayload,
+  defaultRequestConfig,
+  defaultPostRequestConfig,
+  removeUndefinedAndNullValues,
+  isDefinedAndNotNull,
+  simpleProcessRouterDest,
+} = require('../../util');
+const { EventType } = require('../../../constants');
+const {
+  ConfigCategories,
+  mappingConfig,
+  BASE_URL,
+  authHeaderConstant
+} = require('./config');
+
+const { InstrumentationError, OAuthSecretError, ConfigurationError } = require('../../util/errorTypes');
+const { JSON_MIME_TYPE } = require('../../util/constant');
+
+const getOAuthFields = ({ secret }) => {
+  if (!secret) {
+    throw new OAuthSecretError('[TWITTER ADS]:: OAuth - access keys not found');
+  }
+  const oAuthObject = {
+    consumerKey: secret.consumerKey,
+    consumerSecret: secret.consumerSecret,
+    accessToken: secret.accessToken,
+    accessTokenSecret: secret.accessTokenSecret
+  };
+  return oAuthObject;
+};
+
+function getAuthHeaderForRequest(request, oAuthObject) {
+    const oauth = oauth1a({
+      consumer: { key: oAuthObject.consumerKey, secret: oAuthObject.consumerSecret },
+      signature_method: 'HMAC-SHA1',
+      hash_function(base_string, k) {
+        return crypto
+            .createHmac('sha1', k)
+            .update(base_string)
+            .digest('base64')
+      },
+    })
+
+    const authorization = oauth.authorize(request, {
```

Comment:
Why are we doing authorisation here ? Is this not handled on control-plane side itself ?!

Reply:
from control plane we will get token, token secret ... we need to populate Authorization with signature, timestamp and other things as required in OAuth1.0

Reply:
@koladilip are we ok with sending consumer api key and consumer secret from rudder-auth -> transformer ? Can you put your comments on this ?!

Reply:
We are ok with doing this for now. Hence resolving the comment